peer/brontide.go

```diff
@@ -4111,6 +4113,26 @@ func (p *Brontide) handleRemovePendingChannel(req *newChannelMsg) {
 	p.addedChannels.Delete(chanID)
 }
```

Comment:
Please no. Implementation should always be *last*. Interface design first. Use the interface operations in the surrounding code next to validate the ergonomics and structure of the interface design. Finally, implement the interface last.

If you go the other way around it causes you to compromise the interface design in favor of the implementation rather than the call-sites. This will cause the quality of the library code in the codebase to atrophy. Intermediate commits not being fully valid implementations is completely fine. If you absolutely do not want to tolerate invalid intermediate commits, you can squash those commits later.

Work backwards from the goal, not forwards from the implementation.

Reply:
i do understand & agree with the principle but then I think it is important to make sure that it is not possible to call the interface at runtime yet. Cause that will cause nil panics.

Reply:
What's the issue with that in an intermediate commit?